src/System.Text.Json/tests/WritableJsonApiTests.JsonNumberUnitTests.cs

```diff
@@ -0,0 +1,473 @@
+
+            Assert.False(jsonNumber.TryGetInt16(out short shortResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetInt16());
+
+            Assert.False(jsonNumber.TryGetInt32(out int intResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetInt32());
+
+            Assert.False(jsonNumber.TryGetInt64(out long longResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetInt64());
+
+            Assert.False(jsonNumber.TryGetSByte(out sbyte sbyteResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetSByte());
+
+            Assert.False(jsonNumber.TryGetUInt16(out ushort ushortResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetUInt16());
+
+            Assert.False(jsonNumber.TryGetUInt32(out uint uintResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetUInt32());
+
+            Assert.False(jsonNumber.TryGetUInt64(out ulong ulongResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetUInt64());
+        }
+
+        [Theory]
+        [InlineData("0")]
+        [InlineData("-17")]
+        [InlineData("17")]
+        [InlineData("3.14")]
+        [InlineData("-3.1415")]
+        [InlineData("1234567890")]
+        [InlineData("1e400")]
+        [InlineData("1e+100000002")]
+        [InlineData("184467440737095516150.184467440737095516150")]
+        public static void TestToString(string value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetFormattedValue(value);
+            Assert.Equal(value, jsonNumber.ToString());
+        }
+
+        [Fact]
+        public static void TestChangingTypes()
+        {
+            var jsonNumber = new JsonNumber(5);
+            Assert.Equal(5, jsonNumber.GetInt32());
+
+            jsonNumber.SetDouble(3.14);
+            Assert.Equal(3.14, jsonNumber.GetDouble());
+
+            jsonNumber.SetByte(17);
+            Assert.Equal(17, jsonNumber.GetByte());
+
+            jsonNumber.SetInt64(long.MaxValue);
+            Assert.Equal(long.MaxValue, jsonNumber.GetInt64());
+
+            jsonNumber.SetUInt16(ushort.MaxValue);
+            Assert.Equal(ushort.MaxValue, jsonNumber.GetUInt16());
+
+            jsonNumber.SetSingle(-1.1f);
+            Assert.Equal(-1.1f, jsonNumber.GetSingle());
+
+            jsonNumber.SetSByte(4);
+            Assert.Equal(4, jsonNumber.GetSByte());
+
+            jsonNumber.SetUInt32(127);
+            Assert.Equal((uint)127, jsonNumber.GetUInt32());
+
+            jsonNumber.SetFormattedValue("1e400");
+            Assert.Equal("1e400", jsonNumber.ToString());
+
+            jsonNumber.SetUInt64(ulong.MaxValue);
+            Assert.Equal(ulong.MaxValue, jsonNumber.GetUInt64());
+
+            jsonNumber.SetDecimal(decimal.MaxValue);
+            Assert.Equal(decimal.MaxValue, jsonNumber.GetDecimal());
+        }
+
+        [Fact]
+        public static void TestImplicitOperators()
```

Comment:
Add tests for the equality (`.Equals(...)`, `==`, `!=`, and going through the `IEquatable<JsonNumber>` interface), including comparison with `null`.

Also for `GetHashCode()`.

As an aside Can you please measure the code coverage for the new code here and share that?
https://github.com/dotnet/corefx/blob/4b6562e0da17ba1d92d72bc2406af9eea3e9337a/Documentation/building/code-coverage.md
Running the following in the `corefx\src\System.Text.Json\tests\` directory should generate the report.
```text
dotnet msbuild /t:BuildAndTest /p:Coverage=true
```